ksql-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -208,71 +199,133 @@ public double measure(final MetricConfig metricConfig, final long l) {
+    createSensor(
+        metrics,
+        metricName,
+        description,
+        () -> new MeasurableStat() {
           @Override
           public double measure(final MetricConfig metricConfig, final long l) {
             return ksqlEngine.numberOfPersistentQueries();
           }
 
           @Override
           public void record(final MetricConfig metricConfig, final double v, final long l) {
-            // No action for record since we can read the desired results directly.
+            // We don't want to record anything, since the live queries anyway.
           }
         }
     );
   }
 
   private Sensor configureIdleQueriesSensor(final Metrics metrics) {
-    final Sensor sensor = createSensor(metrics, "num-idle-queries");
-    sensor.add(metrics.metricName(
-        ksqlServiceId + "num-idle-queries", this.metricGroupName),
-        new Value());
+    final String metricName = "num-idle-queries";
+    final String description = "Number of inactive queries";
+    final Sensor sensor = createSensor(metrics, metricName, description, Value::new);
     return sensor;
   }
 
   private Sensor configureMessageConsumptionByQuerySensor(final Metrics metrics) {
-    final Sensor sensor = createSensor(
+    final Sensor sensor = createSensor(metrics, "message-consumption-by-query");
+    configureMetric(
         metrics,
-        ksqlServiceId + "message-consumption-by-query");
-    sensor.add(
-        metrics.metricName(ksqlServiceId + "messages-consumed-max", this.metricGroupName),
-        new Max());
+        sensor,
+        "messages-consumed-max",
+        "max msgs consumed by query",
+        Max::new
+    );
+    configureMetric(
+        metrics,
+        sensor,
+        "messages-consumed-min",
+        "min msgs consumed by query",
+        Min::new
+    );
+    configureMetric(
+        metrics,
+        sensor,
+        "messages-consumed-avg",
+        "mean msgs consumed by query",
+        Avg::new
+    );
+    return sensor;
+  }
+
+  private void configureMetric(
+      final Metrics metrics,
+      final Sensor sensor,
+      final String metricName,
+      final String description,
+      final Supplier<MeasurableStat> statSupplier) {
+    // legacy
     sensor.add(
-        metrics.metricName(ksqlServiceId + "messages-consumed-min", this.metricGroupName),
-        new Min());
+        metrics.metricName(ksqlServiceId + metricName, metricGroupName, description),
+        statSupplier.get());
+    // new
     sensor.add(
-        metrics.metricName(ksqlServiceId + "messages-consumed-avg", this.metricGroupName),
-        new Avg());
-    return sensor;
+        metrics.metricName(metricName, ksqlServiceId + metricGroupName, description),
+        statSupplier.get());
   }
 
   private Sensor createSensor(final Metrics metrics, final String sensorName) {
-    final Sensor sensor = metrics.sensor(sensorName);
+    final Sensor sensor = metrics.sensor(metricGroupName + "-" + sensorName);
```

Comment:
This is more than a refactor and actually changes the sensor name, right? For my understanding, what's the sensor name used for?

Reply:
It's just an in-process name for the Sensor. So as long as its uniqueness is preserved its fine to change it.